azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1038,6 +1039,7 @@ public String submitExecutableFlow(ExecutableFlow exflow, String userId)
           Executor executor = activeExecutors.iterator().next();
           // assign only local executor we have
           reference.setExecutor(executor);
+          executorLoader.assignExecutor(executor.getId(), exflow.getExecutionId());
```

Comment:
I have already covered this fix using dispatch method in https://github.com/azkaban/azkaban/pull/500/files